src/transaction.ts

```diff
@@ -49,18 +64,16 @@ export function assembleTransaction(
     // in the tx, so can make simplification of total classic fees for the
     // soroban transaction will be equal to incoming tx.fee + minResourceFee.
     fee: (classicFeeNum + minResourceFeeNum).toString(),
-    memo: raw.memo,
-    networkPassphrase,
-    timebounds: raw.timeBounds,
-    ledgerbounds: raw.ledgerBounds,
-    minAccountSequence: raw.minAccountSequence,
-    minAccountSequenceAge: raw.minAccountSequenceAge,
-    minAccountSequenceLedgerGap: raw.minAccountSequenceLedgerGap,
-    extraSigners: raw.extraSigners,
+    // apply the pre-built Soroban Tx Data from simulation onto the Tx
+    sorobanData: simulation.transactionData,
   });
 
   switch (raw.operations[0].type) {
     case "invokeHostFunction":
+      // @ts-ignore
```

Comment:
not looking through an IDE atm, what was the warning to ignore?

Reply:
Heh, I was hoping this would fly under the radar :flying_saucer: the warning is that `builder.operations` is not defined in the TypeScript (because it's a private property), but this is the best/only way to _remove_ operations from a `TransactionBuilder`. The correct approach, I think, would be to do something like `{ ignoreOperations: true }` as an option to `cloneFrom`, but that would be a "feature request", so this is an acceptable workaround imo, especially given how much better this is than the existing solution anyway.

Reply:
got it. another option for consideration in future, rather than specific to cloning, i've seen a few other builder patterns with a method to clear the builder state also, like `builder.clearOperations()` which then can setup for using `builder.addOperation()` again.

Reply:
Added a comment to clarify!

Reply:
Good point, that may be a cleaner approach! A reset button of sorts.